app/src/tests/components/contents/common/FilterInput.test.tsx

```diff
@@ -0,0 +1,55 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import { FilterInput } from "../../../../app/components/contents/common/FilterInput";
+import userEvent from "@testing-library/user-event";
+
+describe("FilterByName component", () => {
+  it("filtering on input & calls passed in functions", async () => {
+    const setFilterByName = jest.fn();
+    const postFilterAction = jest.fn();
+    const placeholder = "filter by name...";
+    render(<FilterInput setFilter={setFilterByName} postFilterAction={postFilterAction} placeholder={placeholder} />);
+
+    const input = screen.getByPlaceholderText(/filter by name/i);
+
+    userEvent.type(input, "test");
+
+    await waitFor(() => {
+      expect(setFilterByName).toHaveBeenCalledWith("test");
+    });
+    expect(screen.getByPlaceholderText(placeholder)).toBeVisible();
+    expect(postFilterAction).toHaveBeenCalledTimes(1);
+  });
+
+  it("should add classnames to input if passed in", async () => {
+    render(
+      <FilterInput
+        setFilter={jest.fn()}
+        postFilterAction={jest.fn()}
+        placeholder={"placeholder"}
+        inputClassNames="sm:w-[1000px] h-14"
+      />
+    );
+
+    const input = screen.getByPlaceholderText(/placeholder/i);
+    expect(input).toHaveClass("sm:w-[1000px]");
+    expect(input).toHaveClass("h-14");
+  });
+
+  it("should not show reset button if showResetButton false", async () => {
+    render(
+      <FilterInput
+        setFilter={jest.fn()}
+        postFilterAction={jest.fn()}
+        placeholder={"placeholder"}
+        inputClassNames="sm:w-[1000px] h-14"
+      />
+    );
+
+    const filterInput = await screen.findByPlaceholderText(/placeholder/i);
+    userEvent.type(filterInput, "random");
+
+    await waitFor(() => {
+      expect(screen.queryByRole("button", { name: /reset/i })).not.toBeInTheDocument();
+    });
```

Comment:
This would give more confidence if in another of these tests you asserted that the reset button existed - then we know for sure that this expectation is passing for good reason rather than because e.g. the text of the button was changed.

Reply:
ive given up spent way too long cant get to to work isolated... this Is already being tested in 4 other components (manageroles, manageusers, runner, home)